doc/developing.rst

```diff
@@ -4,48 +4,65 @@ Developing Reg
 Install Reg for development
 ---------------------------
 
-First make sure you have virtualenv_ installed for Python 3.5.
+.. highlight:: console
 
-.. _virtualenv: https://pypi.python.org/pypi/virtualenv
-
-Now create a new virtualenv somewhere for Reg's development::
+Clone Reg from github::
 
-  $ virtualenv /path/to/ve_reg
+  $ git clone git@github.com:morepath/reg.git
 
-The goal of this is to isolate you from any globally installed
-versions of setuptools, which may be incompatible with the
-requirements of the buildout tool. You should also be able to recycle
-an existing virtualenv, but this method guarantees a clean one.
+If this doesn't work and you get an error 'Permission denied (publickey)',
+you need to upload your ssh public key to github_.
 
-Clone Reg from github (https://github.com/morepath/reg) and go to the
-reg directory::
+Then go to the reg directory::
 
-  $ git clone git@github.com:morepath/reg.git
   $ cd reg
 
-Now we need to run bootstrap.py to set up buildout, using the Python from the
-virtualenv we've created before::
+Make sure you have virtualenv_ installed.
+
+Create a new virtualenv for Python 3 inside the reg directory::
+
+  $ virtualenv -p $(which python3) env/py3
+
+Activate the virtualenv::
+
+  $ source env/py3/bin/activate
+
+Install the various dependencies and development tools from
+develop_requirements.txt::
+
+  $ pip install -Ur develop_requirements.txt
+
+For upgrading the requirements just run the command again.
+
+If you want to test Reg with Python 2.7 as well you can create a
+second virtualenv for it::
 
-  $ /path/to/ve_reg/bin/python bootstrap.py
+  $ virtualenv -p $(which python2.7) env/py27
 
-This installs buildout, which can now set up the rest of the development
-environment::
+You can then activate it::
 
-  $ bin/buildout
+  $ source env/py27/bin/activate
 
-This will download and install various dependencies and tools.
+and install the develop requirements as described above.
+
+.. note::
+
+   The following commands work only if you have the virtualenv activated.
+
+.. _github: https://help.github.com/articles/generating-an-ssh-key
+
+.. _virtualenv: https://pypi.python.org/pypi/virtualenv
 
 Running the tests
 -----------------
 
-You can run the tests using `py.test`_. Buildout will have installed
-it for you in the ``bin`` subdirectory of your project::
+You can run the tests using `py.test`_::
 
-  $ bin/py.test reg
+  $ py.test
```

Comment:
Since this command and those below assume you are in a virtual environment, you might want to use the ``(py3)`` prefix for the prompt:

```console
(py3) $ py.test
```